File: InsertionSort/InsertionSort.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

void insertionSort(vector<int>& arr) {
    int n = arr.size();
    for (int i = 1; i < n; ++i) {
        int key = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = key;
    }
}

int main() {
    vector<int> arr;
    ifstream infile("random_ints.txt");
    int num;
    while (infile >> num) {
        arr.push_back(num);
    }
    infile.close();

    insertionSort(arr);

    for (int i = 0; i < arr.size(); ++i) {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}

File: InsertionSort/InsertionSort.py
def insertionSort(arr):
    n = len(arr)
    for i in range(1, n):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key

arr = []
with open("random_ints.txt", "r") as infile:
    for line in infile:
        nums = line.split()
        for num in nums:
            arr.append(int(num))

insertionSort(arr)

for i in range(len(arr)):
    print(arr[i], end=" ")
print()
